### PR TITLE
Don't error if multiple UDP row inputs match, unless their output differ

### LINKED ISSUE #926: primitive elaboration too strict
**Describe the bug**
When multiple table rows in a primitive overlap but don't contradict, slang issues an error.

**To Reproduce**
The test below was extracted from a library:
```
 primitive D (q, clk, d, _c, _s);
          output reg q;
          input clk, d, _c, _s;
          table
          // clk in  _c  _s  : Qt  : Qt+1                             
            r    0    1   1  : ?   :  0;
            *    0    ?   1  : 0   :  -;
          endtable
 endprimitive
```
When run, I get:
```
t.v:7:13: error: primitive table row duplicates a set of inputs with a different specified output value
            *    0    ?   1  : 0   :  -;
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
t.v:6:13: note: previous definition here
            r    0    1   1  : ?   :  0;
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
**Additional context**

Section 29.3.4 of IEEE1800-2017 says:

`It shall be illegal to have the same combination of inputs, including edges, specify different output values.`

It doesn't say inputs can't overlap (and when using wildcards it is hard to avoid).  
In the case above, the inputs partially overlap, but don't contradict where they do.  
Regardless of whether this should be reported or now, I think it should be a warning in the worst case.

### PR DESCRIPTION
**Pull request by udif**
This fixes #926 .
It takes into account the 'keep previous state' output value when comparing UDP table rows.
Also includes a new unit test.

### CODE REVIEW

File: source/ast/symbols/MemberSymbols.cpp [hunk 1]
```diff
@@ -995,10 +995,9 @@ static void createTableRow(const Scope& scope, const UdpEntrySyntax& syntax,
         return;
     }
 
-    char stateChar = 0;
-    if (syntax.current) {
-        if (syntax.current->kind == SyntaxKind::UdpSimpleField) {
-            auto raw = syntax.current->as<UdpSimpleFieldSyntax>().field.rawText();
+    auto getstateChar = [](const UdpFieldBaseSyntax* base) -> char {
```
Comment (MikePopoloski) [inline source/ast/symbols/MemberSymbols.cpp:998]: `getstateChar` should be `getStateChar`

File: source/ast/symbols/MemberSymbols.cpp [hunk 2]
```diff
@@ -1007,17 +1006,18 @@ static void createTableRow(const Scope& scope, const UdpEntrySyntax& syntax,
...
                 }
             }
         }
+        return 0;
+    };
 
-        if (!stateChar)
-            return;
-    }
+    auto stateChar = getstateChar(syntax.current);
```
Comment (MikePopoloski) [inline source/ast/symbols/MemberSymbols.cpp:1018]: This is wrong now; not all UDPs have a state character, so I believe this breaks checking for combinational UDPs. I guess that means we're missing a test that fails for dup rows in a combinational UDP.

File: source/ast/symbols/MemberSymbols.cpp (hunk 1 above)
Comment (udif) [reply source/ast/symbols/MemberSymbols.cpp]: Fixed. Somehow this got through.

File: source/ast/symbols/MemberSymbols.cpp (hunk 2 above)
Comment (udif) [reply source/ast/symbols/MemberSymbols.cpp]: You are right. I accidently dropped the check on `syntax.current`.
I added a new test for conflicting outputs on a combinatorial UDP, and indeed the original PR breaks the test.

**Pull request merged**